### PR TITLE
Feature: tools: Use glib for cmdline handling in crm_diff.

### PR DESCRIPTION
**Pull request by clumens**

### CODE REVIEW

**Comment by kgaillot** [comment]
Looks good, but I'll wait until the interface changes make it to master so we can rebase on that

**Comment by clumens** [comment]
This has been updated for master.  Unless I am overlooking something, all that was necessary was adding another NULL to pcmk__build_arg_context.

**Comment by clumens** [comment]
It's a good habit to get into anyway.

File: tools/crm_diff.c [hunk 1]
```diff
@@ -19,44 +19,103 @@
...
+struct {
+    gboolean apply;
+    gboolean as_cib;
+    gboolean no_version;
+    gboolean raw_1;
+    gboolean raw_2;
+    gboolean use_stdin;
+    char *xml_file_1;
+    char *xml_file_2;
+} options;
```
Comment (kgaillot) [inline tools/crm_diff.c]: I think you either have to make this static or init it to { FALSE, } to ensure it's all zeroes to start with

Comment (kgaillot) [reply tools/crm_diff.c]: Never mind, globals are zeroed

**Pull request merged**